pytest-embedded/pytest_embedded/plugin.py

```diff
@@ -1167,12 +1168,34 @@ def dut(
     serial: t.Optional[t.Union['Serial', 'LinuxSerial']],
     qemu: t.Optional['Qemu'],
     wokwi: t.Optional['WokwiCLI'],
+    request: FixtureRequest,
 ) -> t.Union[Dut, t.List[Dut]]:
     """
     A device under test (DUT) object that could gather output from various sources and redirect them to the pexpect
     process, and run `expect()` via its pexpect process.
     """
-    return dut_gn(**locals())
+    dut_instance = dut_gn(
+        _fixture_classes_and_options=_fixture_classes_and_options,
+        openocd=openocd,
+        gdb=gdb,
+        app=app,
+        serial=serial,
+        qemu=qemu,
+        wokwi=wokwi,
+    )
+
+    if _testcase_app_paths_map_key not in request.config.stash:
+        request.config.stash[_testcase_app_paths_map_key] = {}
+
+    stash = request.config.stash[_testcase_app_paths_map_key]
+    testsuite_name = dut_instance.test_case_name
```

Comment:
nitpick. testsuite means something different in junit report field.

usually something like 

```
    <testsuite name="pytest" errors="0" failures="1" skipped="0" tests="1" time="33.576" timestamp="2025-01-23T14:07:31.637710+01:00">
```

the parent concept of all test cases.

you may just call `dut_instance.test_case_name` directly